pkg/systemd/generate/containers.go

```diff
@@ -202,8 +208,11 @@ func containerServiceName(ctr *libpod.Container, options entities.GenerateSystem
 // containerInfo.  Note that the containerInfo is also post processed and
 // completed, which allows for an easier unit testing.
 func executeContainerTemplate(info *containerInfo, options entities.GenerateSystemdOptions) (string, error) {
-	if err := validateRestartPolicy(info.RestartPolicy); err != nil {
-		return "", err
+	if options.RestartPolicy != nil {
+		if err := validateRestartPolicy(*options.RestartPolicy); err != nil {
```

Comment:
we could warn and reset to the default restart policy here, not sure if this makes sense

Reply:
I think we should return an error if the user specified something wrong.  Imagine it being used in a script or some kind of automation. Logs are usually only consulted when something went wrong, so we'd be hiding that.

Reply:
Yes returning an error is better, especially since warnings are not displayed on the remote client.